cypress/integration/ResourcesForResearchers/index.js

```diff
@@ -0,0 +1,264 @@
+    let label;
+    if (section == "Tool Type") {
+      label = "Browse by Tool Type";
+    } else {
+      label = "Browse by Research Area";
+    };
+    const allAnchorTags = document.querySelectorAll(`[aria-label="${label}"]>div>a`);
+    for (let i = 0; i < allAnchorTags; i++) {
+      cy.get('[aria-label="${label}"]>div>a').eq(i).invoke('text').should('match', options[i]);
+    };
+  })
+
+});
+
+Then('the page options should appear at the bottom of the page', () => {
+  cy.get('.r4r-page-options-container').should('be.visible');
+});
+
+And('the following {string} are displayed', (pageOption) => {
+  const options = document.querySelectorAll(`div#PageOptionsControl1 > ul`);
+  for (let i = 0; i < options; i++) {
+    cy.get('div#PageOptionsControl1 > ul>li>a').should('contain.text', title);
+  }
+});
+
+And("user clicks on {string} link", (viewAll) => {
+  cy.get('.r4r-DEFAULT.view-all__link').click();
+});
+
+And("user is redirected to {string} with {string}", (redirectedPath, urlSearch) => {
+  cy.location('pathname').should('eq', redirectedPath);
+  cy.location('search').should('include', urlSearch);
+});
+
+And("{string} link has a href {string}", (linkText, linkHref) => {
+  cy.get('header > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+And("search box is empty", () => {
+  cy.get('.searchbar__container input').should('contain.text', "");
+});
+
+And('results info text is {string}', (resultsInfo) => {
+  const regex = new RegExp(resultsInfo);
+  cy.get('[aria-label="Results count"] p').invoke('text').should('match', regex);
+});
+
+And('both pagers are displayed', () => {
+  cy.get('.r4r-results.r4r-DEFAULT>nav').should('have.length', 2);
+  cy.get('.r4r-results.r4r-DEFAULT>nav').each(($el) => {
+    expect('div r4r-pager__count  r4r-DEFAULT').to.exist;
+    expect('div r4r-pager__nav  r4r-DEFAULT').to.exist;
+  });
+
+});
+
+And('the results are displayed with each title containing a link', () => {
+  cy.get('[aria-label="search results"]').should('be.visible');
+  cy.get('[aria-label="search results"] > article>h2').first().should('be.visible').find('a').should('have.attr', 'href');
+});
+
+And('each result has a description text under the title', () => {
+  cy.get('[aria-label="search results"] > article>p').first().should('be.visible');
+});
+
+When('user clicks on {string}', (tool) => {
+  cy.get('.home-nav__main >div >div>a').contains(tool).click();
+});
+
+And('the filter displays the {string}', (filter) => {
+  cy.get('[aria-label="Selected Search Filters"] [tabindex] p').should('contain.text', filter);
+});
+
+And('the {string} box displays the filter {string}', (section, tool) => {
+  const regex = new RegExp(tool);
+  cy.get('.facet__box--tool-types').should('be.visible').find('h4').should('contain.text', section);
+  cy.get('.r4r-DEFAULT.facet__box.facet__box--tool-types > label[role="checkbox"] > .r4r-DEFAULT.filter__label').invoke('text').should('match', regex);
```

Comment:
this locator is waaay too long and it's first part points to the first cy.get locator , so alias again)

```
cy.get('div[class*="box--tool-types"]')
    .as("toolType")
    .find("h4")
    .should("be.visible")
    .and("contain.text", section);
  cy.get("@toolType").find("span").invoke("text").should("match", regex);
```